### ISSUE TITLE
Differences in flex calculations

### ISSUE THREAD

**Issue by e-oz** [angular/material#2675]
As promised, here is the codepen with reproduced bug:
http://codepen.io/evgeniyoz/pen/mJJQwG
and same as gist: https://gist.github.com/jamm/1944ae86a8818df94406

Issue: flex65+flex35 != flex65+flex
(65 and 35 are just for example, the point is 65+35=100).

**Comment by ThomasBurleson** [comment]
@jamm - we will look at this in 0.10.

**Comment by rschmukler** [comment]
@ThomasBurleson this is actually the same issue as #2681, but with the `margin-left` and `margin-right` being the culprit.

Check out [this codepen](http://codepen.io/rschmukler/pen/rVVbEQ)

which adds the following css rules:

```css
[layout="row"] > [flex] {
  margin-left: auto;
  margin-right: auto;
}

[layout="column"] > [flex] {
  margin-top: auto;
  margin-bottom: auto;
}
```

Which makes the example work as you would expect.  I haven't thought about edge cases enough to recommend that we apply that as a blanket wide solution, but it might be correct. Something to discuss tomorrow.

**Comment by ThomasBurleson** [comment]
Closed with SHA 2478f1d2b064d5e06ba57d01b5460fe232f4b270
